src/cpu/aarch64/jit_uni_reorder.cpp

```diff
@@ -161,20 +161,13 @@ struct jit_uni_reorder_kernel_f32_t : public kernel_t, public jit_generator {
     static bool applicable(const prb_t &p) {
         using namespace data_type;
 
-        bool bf16_ok
-                = (mayiuse_bf16() && (p.itype == bf16) && (p.otype == bf16)
-                          && !interim_f32_needed(p, false) && p.beta == 0.f)
-                || (p.itype != bf16 && p.otype != bf16)
-                || (p.itype == f32 && p.otype == bf16 && mayiuse_bf16()
-                        && p.beta == 0.f);
-
         bool ok = true && p.ndims > 0
-                && utils::one_of(p.itype, f32, bf16, s32, data_type::s8, u8)
+                && utils::one_of(p.itype, f32, s32, data_type::s8, u8)
                 && utils::one_of(p.otype, f32, bf16, s32, data_type::s8, u8)
                 && utils::everyone_is(0, p.ioff, p.ooff) /* do we need this? */
                 && utils::one_of(p.beta, 0.f, 1.f) /* anything else? */
                 && simple_impl_desc_init(p, nullptr) && prb_has_small_strides(p)
-                && bf16_ok;
+                && ((p.otype != bf16) || (p.itype == f32 && mayiuse_bf16()));
```

Comment:
```suggestion
                && IMPLICATION(p.otype == bf16, p.itype == f32 && mayiuse_bf16());
```

Reply:
Thanks for the suggestion, I have applied this in the commit.

Reply:
@dzarukin Hello, we have made the relevant changes, kindly check and support for Merger.
Thanks !